### PR TITLE
Change `sha1` to `sha256` inside `unsign` method

### PR DESCRIPTION
**Pull request by rcalfredson**
Purpose: to use a more secure encryption method in the string-comparison operation inside the `unsign` method. My organization uses the module, and the presence of `sha1` here triggered a security alert, so I wanted to see if this change could be made, please.

### CODE REVIEW

**Comment by natevw** [comment]
I understand this can trigger an automated warning, but to be clear the purpose of the `sha1` for string comparison is to avoid timing attacks — its strength as a cryptographic hash should not be important. 

I think it's time to move to [crypto.timingSafeEqual](https://nodejs.org/dist/latest-v8.x/docs/api/crypto.html#crypto_crypto_timingsafeequal_a_b) rather than working around this false warning — it will make the code much more readable than another hash function.

Basically we need to switch to that, and add `"engines" : { "node" : ">=6.6.0" }` to package.json. Is that something you'd have a chance to tackle?

**Comment by jodevsa** [comment]
What is the reason of comparing their hash values? 

computing hash, requires traversing the 2 strings; this is what is done with the equal operator!!

**Comment by jodevsa** [comment]
@natevw  Done ! #24

**Comment by natevw** [comment]
Closing this, since #24 has been merged and will also avoid any `sha1` warnings. Thanks for getting the ball rolling on this though!
